### PR TITLE
8258839: Remove JVM option ExitVMOnVerifyError

### PR DESCRIPTION
**Pull request by hseigel**
Please review this change to replace the JVM ExitVMOnVerifyError option with asserts.  The change was tested with tiers 1 and 2 on Linux, Mac OS, and Windows, and tiers 3-5 on Linux x64.

Thanks, Harold

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258839](https://bugs.openjdk.java.net/browse/JDK-8258839): Remove JVM option ExitVMOnVerifyError

### Reviewers
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**) ⚠️ Review applies to 9397d7f2aab426da4e222842a713e9866da271cd
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1882/head:pull/1882`
`$ git checkout pull/1882`

### CODE REVIEW

**Review by iignatev — APPROVED** [review]
👍

File: src/hotspot/share/c1/c1_Runtime1.cpp [hunk 1]
```diff
@@ -532,11 +532,9 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
...
   assert(exception.not_null(), "NULL exceptions should be handled by throw_exception");
-  // Check that exception is a subclass of Throwable, otherwise we have a VerifyError
-  if (!(exception->is_a(SystemDictionary::Throwable_klass()))) {
-    if (ExitVMOnVerifyError) vm_exit(-1);
-    ShouldNotReachHere();
-  }
+  // Check that exception is a subclass of Throwable
+  assert(exception->is_a(SystemDictionary::Throwable_klass()),
+         "Exception not subclass of Throwable");
 #endif
```
Comment (dcubed-ojdk) [inline src/hotspot/share/c1/c1_Runtime1.cpp:539]: The "#endif" can be deleted.

File: src/hotspot/share/c1/c1_Runtime1.cpp [hunk 2]
```diff
@@ -532,11 +532,9 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
   }
 #ifdef ASSERT
   assert(exception.not_null(), "NULL exceptions should be handled by throw_exception");
```
Comment (dcubed-ojdk) [inline src/hotspot/share/c1/c1_Runtime1.cpp:534]: The "#ifdef ASSERT" can be deleted.

File: src/hotspot/share/jvmci/jvmciRuntime.cpp [hunk 3]
```diff
@@ -260,11 +260,9 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
 #ifdef ASSERT
   assert(exception.not_null(), "NULL exceptions should be handled by throw_exception");
```
Comment (dcubed-ojdk) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:261]: The "#fdef ASSERT" can be deleted.

File: src/hotspot/share/jvmci/jvmciRuntime.cpp [hunk 4]
```diff
@@ -260,11 +260,9 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
...
   assert(oopDesc::is_oop(exception()), "just checking");
-  // Check that exception is a subclass of Throwable, otherwise we have a VerifyError
-  if (!(exception->is_a(SystemDictionary::Throwable_klass()))) {
-    if (ExitVMOnVerifyError) vm_exit(-1);
-    ShouldNotReachHere();
-  }
+  // Check that exception is a subclass of Throwable
+  assert(exception->is_a(SystemDictionary::Throwable_klass()),
+         "Exception not subclass of Throwable");
 #endif
```
Comment (dcubed-ojdk) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:267]: The "#endif" can be deleted.

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]

**Comment by dcubed-ojdk** [comment]
Minor additional cleanups.

**Review by coleenp — APPROVED** [review]
+1 to Dan's comments.

**Comment by hseigel** [comment]
Please review this updated change that removes the unneeded #ifdef and #endif statements.
Thanks, Harold

File: src/hotspot/share/c1/c1_Runtime1.cpp (hunk 1 above)
Comment (hseigel) [reply src/hotspot/share/c1/c1_Runtime1.cpp]: deleted

File: src/hotspot/share/c1/c1_Runtime1.cpp (hunk 2 above)
Comment (hseigel) [reply src/hotspot/share/c1/c1_Runtime1.cpp]: deleted

File: src/hotspot/share/jvmci/jvmciRuntime.cpp (hunk 3 above)
Comment (hseigel) [reply src/hotspot/share/jvmci/jvmciRuntime.cpp:261]: deleted

File: src/hotspot/share/jvmci/jvmciRuntime.cpp (hunk 4 above)
Comment (hseigel) [reply src/hotspot/share/jvmci/jvmciRuntime.cpp:267]: deleted

**Comment by hseigel** [comment]
Thanks Igor, Dan and Coleen for the reviews and for pointing out the "#..." statements that could be deleted.
Harold

File: src/hotspot/share/jvmci/jvmciRuntime.cpp (hunk 3 above)
Comment (dcubed-ojdk) [reply src/hotspot/share/jvmci/jvmciRuntime.cpp:261]: This one is not yet fixed.

File: src/hotspot/share/jvmci/jvmciRuntime.cpp (hunk 4 above)
Comment (dcubed-ojdk) [reply src/hotspot/share/jvmci/jvmciRuntime.cpp:267]: This one is not yet fixed.

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]

**Comment by dcubed-ojdk** [comment]
One file was fixed, but the other was not.

**Comment by hseigel** [comment]
Ugh!  I forgot to stage the file.

**Review by coleenp — APPROVED** [review]

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up!

**Comment by hseigel** [comment]
Thanks again for the reviews!

/integrate

**Pull request closed without merging**